.github/workflows/s3_test.yml

```diff
@@ -17,17 +17,17 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-latest]
+        os: [macos-latest, ubuntu-latest]
 
     steps:
     - uses: actions/checkout@v2
       with:
         ref: refs/pull/${{ github.event.pull_request.number }}/merge
         submodules: recursive
-    - name: Set up Python 3.7
+    - name: Set up Python 3.10
```

Comment:
We can consider running the tests on multiple Python versions.

Reply:
Yes good idea.

I was debugging why the tests were erroring out for me, hence trying this 3.10 thing out.  Turns out it was something else.

I will close this draft.